### DISCUSSION TITLE
How to reference current item when style is applied

### DISCUSSION THREAD

**Discussion by sam-wheat** [dotnet/wpf#6431] [Q&A] [answered]
In the XAML below I want to pass the `ListBoxItem` to which the style is being applied to the converter.  What I am getting is an instance of `MainWindow`.  Why is this true, and how do I reference the `ListBoxItem` to which the style is being applied?
For the record, this problem is mostly academic.  I am aware the converter is unnecessary - I am trying to solve a different problem and ran into this along the way.

Example project attached.

        <Style TargetType="ListBoxItem" x:Key="zzz">
                <Setter Property="IsSelected">
                    <Setter.Value>
                        <MultiBinding Converter="{StaticResource SelectedItemToBoolConverter}">
                            <Binding Path="." />
                            <Binding RelativeSource="{RelativeSource AncestorType=ListBox}" Path="." />
                        </MultiBinding>
                    </Setter.Value>
                </Setter>
                <Setter Property="Template">
                    <Setter.Value>
                        <ControlTemplate TargetType="ListBoxItem">
                            <CheckBox Margin="5,2" IsChecked="{TemplateBinding IsSelected}">
                                <ContentPresenter />
                            </CheckBox>
                        </ControlTemplate>
                    </Setter.Value>
                </Setter>
            </Style>
        </StackPanel.Resources>

        <ListBox ItemContainerStyle="{StaticResource zzz}" >
            <ListBox.Items>
                <ListBoxItem Content="Item 1"/>
                <ListBoxItem Content="Item 2"/>
                <ListBoxItem Content="Item 3"/>
                <ListBoxItem Content="Item 4"/>
                <ListBoxItem Content="Item 5"/>
            </ListBox.Items>
        </ListBox>

Converter:

    public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
     {
         ListBoxItem lbi = values[0] as ListBoxItem; // Expected: ListBoxItem  Actual: MainWindow
         ListBox lb = values[1] as ListBox; 
         return lbi == lb.SelectedItem;
     }

[CommandProblem3.zip](https://github.com/dotnet/wpf/files/8489350/CommandProblem3.zip)

**Comment by batzen** [comment, marked as answer]
You have to bind to RelativeSource=Self instead of ".". "." references to the current DataContext instead of an UI element.

**Reply by sam-wheat** [reply]
Thank you @batzen!
